commands/core/drupal/update.inc

```diff
@@ -130,6 +130,11 @@ function update_main() {
   drupal_load_updates();
   update_fix_compatibility();
 
+  // Check requirements before updating.
+  if (!drush_update_check_requirements()) {
```

Comment:
Well then we have to go through the rigmarole of loading all the .install files in the validation